### ISSUE TITLE
Can't train QLoRA.

### ISSUE THREAD

**Issue by nivibilla** [lvwerra/trl#552]
I'm using this tutorial, 
https://medium.com/@ud.chandra/instruction-fine-tuning-llama-2-with-pefts-qlora-method-d6a801ebb19

But when I run the code I get . The only change I've made is to enable gradient checkpointing.

`Expected to mark a variable ready only once.`

**Comment by ahron1** [comment]
Does it work without gradient checkpointing?

**Comment by nivibilla** [comment]
issue was with databricks. not trl
